### PR TITLE
Fix possible Ingress port collisions

### LINKED ISSUE #1680: Add-On port conflict - nginx: [emerg] bind() to 172.30.32.1:64386 failed (98: Address in use)
**Home Assistant release with the issue: 0.108.8**

**Operating environment (HassOS/Generic): Generic - Raspbian GNU/Linux 10 (buster)**

**Supervisor logs:**

**Description of problem:**
Two Add-Ons are unable to run simultaneously due to a port conflict.  Add-On port numbers are randomly generated and I won the lottery.  :/  Restarting Add-Ons, Uninstall/Re-Installing Add-Ons, Reloading Supervisor has no effect.

Starting AdGuard Home if Node-RED is running - After "INFO: Starting NGix..." I see:
```
2020/04/24 11:46:43 [emerg] 410#410: bind() to 172.30.32.1:64386 failed (98: Address in use)
nginx: [emerg] bind() to 172.30.32.1:64386 failed (98: Address in use)
```
It retries 5 times then fails and stops.

If I stop the Node-RED Add-On, then start the AdGuard Home Add-On it comes up fine.
Starting Node-RED if AdGuard Home is running - 
```
2020/04/24 11:52:25 [emerg] 443#443: bind() to 172.30.32.1:64386 failed (98: Address in use)
nginx: [emerg] bind() to 172.30.32.1:64386 failed (98: Address in use)
```

Based on a Discord chat with @frenck and @paul_sinclair I looked in **/usr/share/hassio/ingress.json** and it contains the following:

```
{
  "ports": {
    "a0d7b954_nodered": 64386,
    "15ef4d2f_esphome": 63642,
    "a0d7b954_motioneye": 62271,
    "a0d7b954_adguard": 64386
  },
```

Attempting to edit the ingress.json to remove the adguard line, edit the port number so there is no conflict, and restarting the supervisor or uninstalling/reinstalling the Add-On does not fix the problem nor change the symptom - the Add-On will still fail to start if the other is running and interestingly the port number doesn't change in the logs, they remain as stated above.

### PR DESCRIPTION
**Pull request by frenck**
Adds a small loop to keep generating random Ingress ports until a unique one is found.

Fixes #1680

### CODE REVIEW

**Comment by frenck** [comment]
Converted to draft, wanted to add another thing

**Comment by pvizeli** [comment]
We have a function they can check if a port is in use or not on the gateway address

File: supervisor/ingress.py [hunk 1]
```diff
@@ -120,7 +122,11 @@ def get_dynamic_port(self, addon_slug: str) -> int:
             return self.ports[addon_slug]
 
         port = None
-        while port is None or port in self.ports.values():
+        while (
+            port is None
+            or port in self.ports.values()
+            or check_port(IPv4Address(self.coresys.docker.network.gateway), port)
```
Comment (pvizeli) [inline supervisor/ingress.py]:
```suggestion
            or check_port(self.sys_docker.network.gateway, port)
```

File: supervisor/ingress.py [hunk 2]
```diff
@@ -118,13 +119,28 @@ def get_dynamic_port(self, addon_slug: str) -> int:
         """Get/Create a dynamic port from range."""
         if addon_slug in self.ports:
             return self.ports[addon_slug]
-        port = random.randint(62000, 65500)
+
+        port = None
+        while (
+            port is None
+            or port in self.ports.values()
+            or check_port(self.coresys.docker.network.gateway, port)
```
Comment (pvizeli) [inline supervisor/ingress.py:127]:
```suggestion
            or check_port(self.sys_docker.network.gateway, port)
```

**Pull request merged**